ibm/resource_ibm_app_config_environment.go

```diff
@@ -0,0 +1,258 @@
+				Computed:    true,
+				Description: "Creation time of the environment.",
+			},
+			"updated_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last modified time of the environment data.",
+			},
+			"href": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Environment URL.",
+			},
+		},
+	}
+}
+
+func getAppConfigClient(meta interface{}, guid string) (*appconfigurationv1.AppConfigurationV1, error) {
+	appconfigClient, err := meta.(ClientSession).AppConfigurationV1()
+	if err != nil {
+		return nil, err
+	}
+	bluemixSession, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return nil, err
+	}
+	appConfigURL := fmt.Sprintf("https://%s.apprapp.cloud.ibm.com/apprapp/feature/v1/instances/%s", bluemixSession.Config.Region, guid)
+	url := envFallBack([]string{"IBMCLOUD_APP_CONFIG_API_ENDPOINT"}, appConfigURL)
+	appconfigClient.Service.Options.URL = url
+	return appconfigClient, nil
+}
+
+func resourceEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	guid := d.Get("guid").(string)
+	appconfigClient, err := getAppConfigClient(meta, guid)
+	if err != nil {
+		return err
+	}
+	options := &appconfigurationv1.CreateEnvironmentOptions{}
+
+	options.SetName(d.Get("name").(string))
+	options.SetEnvironmentID(d.Get("environment_id").(string))
+	if _, ok := d.GetOk("description"); ok {
+		options.SetDescription(d.Get("description").(string))
+	}
+	if _, ok := d.GetOk("tags"); ok {
+		options.SetTags(d.Get("tags").(string))
+	}
+	if _, ok := d.GetOk("color_code"); ok {
+		options.SetColorCode(d.Get("color_code").(string))
+	}
+	_, response, err := appconfigClient.CreateEnvironment(options)
+
+	if err != nil {
+		return fmt.Errorf("[DEBUG] CreateEnvironment failed %s\n%s", err, response)
+	}
+	d.SetId(fmt.Sprintf("%s/%s", guid, *options.EnvironmentID))
+
+	return resourceEnvironmentRead(d, meta)
+}
+
+func resourceEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	if ok := d.HasChanges("name", "tags", "color_code", "description"); ok {
+		parts, err := idParts(d.Id())
+		if err != nil {
+			return nil
+		}
+		appconfigClient, err := getAppConfigClient(meta, parts[0])
+		if err != nil {
+			return err
+		}
+
+		options := &appconfigurationv1.UpdateEnvironmentOptions{}
+
+		options.SetName(d.Get("name").(string))
+		options.SetEnvironmentID(d.Get("environment_id").(string))
+		if _, ok := d.GetOk("description"); ok {
+			options.SetDescription(d.Get("description").(string))
+		}
+		if _, ok := d.GetOk("tags"); ok {
+			options.SetTags(d.Get("tags").(string))
+		}
+		if _, ok := d.GetOk("color_code"); ok {
+			options.SetColorCode(d.Get("color_code").(string))
+		}
+
+		_, response, err := appconfigClient.UpdateEnvironment(options)
+		if err != nil {
+			d.SetId("")
+			return fmt.Errorf("[DEBUG] UpdateEnvironment failed %s\n%s", err, response)
+		}
+		d.SetId(fmt.Sprintf("%s/%s", parts[0], *options.EnvironmentID))
```

Comment:
No need to set ID in update method...we need to set only in create

Reply:
updated